# Toplulaştırma ve Gruplama ( Aggregation & Grouping )
---
**Basit toplulaştırma fonksiyonları :**

* count()
* first()
* last()
* mean()
* median()
* min()
* max()
* std()
* var()
* sum()
---

## Toplulaştırma İşlemleri - AGGREGATİON 

In [8]:
import seaborn as sns 



import warnings
warnings.filterwarnings("ignore")
warnings.warn("this will not show")

In [3]:
?sns.load_dataset

### Seaborn üzerinde hazır veri setleri mevcut.
---
[Mevcut veri setlerine ulaşmak için tıklayınız.  ](https://github.com/mwaskom/seaborn-data)
--

In [4]:
df = sns.load_dataset('planets')
df

method  number  orbital_period   mass  distance  year
0     Radial Velocity       1      269.300000   7.10     77.40  2006
1     Radial Velocity       1      874.774000   2.21     56.95  2008
2     Radial Velocity       1      763.000000   2.60     19.84  2011
3     Radial Velocity       1      326.030000  19.40    110.62  2007
4     Radial Velocity       1      516.220000  10.50    119.47  2009
...               ...     ...             ...    ...       ...   ...
1030          Transit       1        3.941507    NaN    172.00  2006
1031          Transit       1        2.615864    NaN    148.00  2007
1032          Transit       1        3.191524    NaN    174.00  2007
1033          Transit       1        4.125083    NaN    293.00  2008
1034          Transit       1        4.187757    NaN    260.00  2008

[1035 rows x 6 columns]

In [5]:
df.head()

method  number  orbital_period   mass  distance  year
0  Radial Velocity       1         269.300   7.10     77.40  2006
1  Radial Velocity       1         874.774   2.21     56.95  2008
2  Radial Velocity       1         763.000   2.60     19.84  2011
3  Radial Velocity       1         326.030  19.40    110.62  2007
4  Radial Velocity       1         516.220  10.50    119.47  2009

In [6]:
df.shape

(1035, 6)

---
>> **Yukarıda gözlemlendiği üzere : Planets Veri setimiz 1035 gözlem ve altı değişkenden oluşmaktadır!**
---

In [9]:
df.mean()

number               1.785507
orbital_period    2002.917596
mass                 2.638161
distance           264.069282
year              2009.070531
dtype: float64

In [10]:
df.min()

method            Astrometry
number                     1
orbital_period      0.090706
mass                  0.0036
distance                1.35
year                    1989
dtype: object

In [12]:
# Mass değişkeni için : En küçük değer
df['mass'].min()

0.0036

In [13]:
# Mass değişkeni için : En büyük değer
df['mass'].max()

25.0

In [15]:
#Sayma
df['mass'].count()

513

In [16]:
#Ortalama
df['mass'].mean()

2.6381605847953216

In [17]:
#Toplam
df['mass'].sum()

1353.37638

In [18]:
#standart sapma 
df['mass'].std()

3.8186166509616046

In [19]:
#Varyans
df['mass'].var()

14.58183312700122

### Describe fonksiyonu ile elimizdeki verileri betimliyoruz! 

In [22]:
#Bütün değişkenler için betimsel istatistikler temel , dağılım istatistikleri 
df.describe()

number  orbital_period        mass     distance         year
count  1035.000000      992.000000  513.000000   808.000000  1035.000000
mean      1.785507     2002.917596    2.638161   264.069282  2009.070531
std       1.240976    26014.728304    3.818617   733.116493     3.972567
min       1.000000        0.090706    0.003600     1.350000  1989.000000
25%       1.000000        5.442540    0.229000    32.560000  2007.000000
50%       1.000000       39.979500    1.260000    55.250000  2010.000000
75%       2.000000      526.005000    3.040000   178.500000  2012.000000
max       7.000000   730000.000000   25.000000  8500.000000  2014.000000

In [23]:
#Bütün değişkenler için betimsel istatistikler temel , dağılım istatistikleri 
#Tersini almak istediğimiz zaman fonksiyonun sonuna .T Ekliyoruz
#.T -> Transpozunu almak anlamına gelir!
df.describe().T

count         mean           std          min         25%  \
number          1035.0     1.785507      1.240976     1.000000     1.00000   
orbital_period   992.0  2002.917596  26014.728304     0.090706     5.44254   
mass             513.0     2.638161      3.818617     0.003600     0.22900   
distance         808.0   264.069282    733.116493     1.350000    32.56000   
year            1035.0  2009.070531      3.972567  1989.000000  2007.00000   

                      50%       75%       max  
number             1.0000     2.000       7.0  
orbital_period    39.9795   526.005  730000.0  
mass               1.2600     3.040      25.0  
distance          55.2500   178.500    8500.0  
year            2010.0000  2012.000    2014.0

In [24]:
#----------------------------------------------------------------------

In [25]:
#Örnek : 

df.dropna().describe().T

count         mean          std        min         25%  \
number          498.0     1.734940     1.175720     1.0000     1.00000   
orbital_period  498.0   835.778671  1469.128259     1.3283    38.27225   
mass            498.0     2.509320     3.636274     0.0036     0.21250   
distance        498.0    52.068213    46.596041     1.3500    24.49750   
year            498.0  2007.377510     4.167284  1989.0000  2005.00000   

                     50%        75%      max  
number             1.000     2.0000      6.0  
orbital_period   357.000   999.6000  17337.5  
mass               1.245     2.8675     25.0  
distance          39.940    59.3325    354.0  
year            2009.000  2011.0000   2014.0

---
### Yukarıda gözlemlendiği üzere bazı veirlerde ufak-tefek değişiklik yaşandı ! 
---
* Eksik değerler silinerek betimleme işlemi gerçekleşti!**
* Ancak bu doğru bir kullanım değildir. Örnek olması bakımında ele alınmıştır.